types/node/v16/crypto.d.ts

```diff
@@ -3329,8 +3371,522 @@ declare module 'crypto' {
+                extractable: boolean, keyUsages: ReadonlyArray<CryptoKeyUsages>
+            ): Promise<CryptoKey>;
+            /**
+             * Using the method and parameters given in `algorithm` and the keying material provided by `key`,
+             * `subtle.verify()` attempts to verify that `signature` is a valid cryptographic signature of `data`.
+             * The returned promise is resolved with either `true` or `false`.
+             *
+             * The algorithms currently supported include:
+             *
+             * - `'RSASSA-PKCS1-v1_5'`
+             * - `'RSA-PSS'`
+             * - `'ECDSA'`
+             * - `'HMAC'`
+             * - `'NODE-DSA'`
+             * - `'NODE-ED25519'`
+             * - `'NODE-ED448'`
+             * @since v15.0.0
+             */
+            verify(algorithm: SubtleCrypto.VerifyAlgorithm, key: CryptoKey, signature: BufferSource, data: BufferSource): Promise<boolean>;
+            /**
+             * In cryptography, "wrapping a key" refers to exporting and then encrypting the keying material.
+             * The `subtle.wrapKey()` method exports the keying material into the format identified by `format`,
+             * then encrypts it using the method and parameters specified by `wrapAlgo` and the keying material provided by `wrappingKey`.
+             * It is the equivalent to calling `subtle.exportKey()` using `format` and `key` as the arguments,
+             * then passing the result to the `subtle.encrypt()` method using `wrappingKey` and `wrapAlgo` as inputs.
+             * If successful, the returned promise will be resolved with an `<ArrayBuffer>` containing the encrypted key data.
+             *
+             * The wrapping algorithms currently supported include:
+             *
+             * - `'RSA-OAEP'`
+             * - `'AES-CTR'`
+             * - `'AES-CBC'`
+             * - `'AES-GCM'`
+             * - `'AES-KW'`
+             * @param format Must be one of `'raw'`, `'pkcs8'`, `'spki'`, or `'jwk'`.
+             * @since v15.0.0
+             */
+            wrapKey(format: SubtleCrypto.KeyFormat, key: CryptoKey, wrappingKey: CryptoKey, wrapAlgo: SubtleCrypto.WrappingAlgorithm): Promise<ArrayBuffer>;
+        }
+        namespace SubtleCrypto {
+            type KeyFormat = 'raw' | 'pkcs8' | 'spki' | 'jwk';
+            type KeyFormatNodeJS = KeyFormat | 'node.keyObject';
+            type EcNamedCurve = 'P-256' | 'P-384' | 'P-521' | 'NODE-ED25519' | 'NODE-ED448' | 'NODE-X25519' | 'NODE-X448';
+            type EcParamsNames = EcdsaParams['name'] | EcdhKeyDeriveParams['name'];
+            type AesParamsNames = AesCbcParams['name'] | AesCtrParams['name'] | AesGcmParams['name'] | AesKwParams['name'];
+            type RsaParamsNames = RsaOaepParams['name'] | RsaPssParams['name'] | RsaSignParams['name'];
+            type RsaOaepAesParams = RsaOaepParams | AesCtrParams | AesCbcParams | AesGcmParams;
```

Comment:
This name is not very obvious and I initially read it as `RsaOaepParams`. Would `EncryptionAlgorithmParams` prevent this and also allow for easier extension e.g. if it introduces quantum safe crypto encryption algorithms.

Reply:
Resolved by this PR: https://github.com/jhmaster2000/DefinitelyTyped/pull/2